src/reduxSetup/reducers/popularMenuReducer.js

```diff
@@ -1,10 +1,18 @@
-import { POPULAR_MENU } from "../constants";
-const initialState = {popularMenu: null}
+import { POPULAR_MENU, SPECIAL_MENU } from "../constants";
+const initialState = {popularMenu: null, specialMenu:null}
 export const popularMenuReducer = (state = initialState, action) => {
     switch (action.type) {
         case POPULAR_MENU:
             return{...state, popularMenu: action.payload};    
         default:
             return state;
     }
+};
+export const specialMenuReducer = (state = initialState, action) => {
```

Comment:
Why have a separate reducer for special menu? why not combine polpuarMenu and special menu into one reducer?
For example
```
export const menuReducer = (state = initialState, action) => {
    switch (action.type) {
        case POPULAR_MENU:
            return{...state, popularMenu: action.payload};    
        case SPECIAL_MENU:
            return{...state, specialMenu: action.payload};
        default:
            return state;
    }
};
```